app/src/main/java/org/mozilla/fenix/settings/address/view/AddressEditorView.kt

```diff
@@ -143,7 +164,28 @@ class AddressEditorView(
         }.show()
     }
 
-    private fun bindCountryDropdown() {
+    private fun bindSubregionDropdown(country: Country) {
+        val subregions = country.subregions
+        val selectedSubregion = address?.addressLevel1?.takeIf { it in subregions }
+            ?: subregions.first()
+
+        val adapter = ArrayAdapter<String>(
+            binding.root.context,
+            android.R.layout.simple_spinner_dropdown_item,
+        )
+
+        var selectedPosition = 0
+        subregions.forEachIndexed { index, subregion ->
+            if (subregion == selectedSubregion) selectedPosition = index
+            adapter.add(subregion)
+        }
```

Comment:
I guess I was operating under the assumption that array adapters implemented some kind of recycler system, so that the position in a list wouldn't necessarily map to the position in the adapter. I'm trying to look up whether that's the case, but do you know for sure the lists never get out of sync?

Reply:
There should still be a recycler system, but the selection index stuff should work as normal. My understanding is RecyclerViews have two running index systems (per your hesitation), but setting the selection index based on the underlying data collection will work as intended.

Reply:
Oh yeah, I guess that was just a fundamental misunderstanding on my part of the purpose of array adapters. Thanks for pointing it out!